SU2_CFD/include/driver_structure.hpp

```diff
@@ -1583,6 +1406,106 @@ class CSinglezoneDriver : public CDriver {
 
 };
 
+/*!
+ * \class CDiscAdjSinglezoneDriver
+ * \brief Class for driving single-zone adjoint solvers.
+ * \author R. Sanchez
+ * \version 6.2.0 "Falcon"
+ */
+class CDiscAdjSinglezoneDriver : public CSinglezoneDriver {
+protected:
+
+  unsigned long nAdjoint_Iter;                  /*!< \brief The number of adjoint iterations that are run on the fixed-point solver.*/
+  unsigned short RecordingState;                /*!< \brief The kind of recording the tape currently holds.*/
+  unsigned short MainVariables,                 /*!< \brief The kind of recording linked to the main variables of the problem.*/
+                 SecondaryVariables;            /*!< \brief The kind of recording linked to the secondary variables of the problem.*/
+  su2double ObjFunc;                            /*!< \brief The value of the objective function.*/
+  CIteration* direct_iteration;                 /*!< \brief A pointer to the direct iteration.*/
+
+  CConfig *config;                              /*!< \brief Definition of the particular problem. */
+  CIteration *iteration;                        /*!< \brief Container vector with all the iteration methods. */
+  CIntegration **integration;                   /*!< \brief Container vector with all the integration methods. */
+  CGeometry *geometry;                          /*!< \brief Geometrical definition of the problem. */
+  CSolver **solver;                             /*!< \brief Container vector with all the solutions. */
+
+
+public:
+
+  /*!
+   * \brief Constructor of the class.
+   * \param[in] confFile - Configuration file name.
+   * \param[in] val_nZone - Total number of zones.
+   * \param[in] MPICommunicator - MPI communicator for SU2.
+   */
+  CDiscAdjSinglezoneDriver(char* confFile,
+             unsigned short val_nZone,
+             unsigned short val_nDim,
+             bool val_periodic,
```

Comment:
Can you please remove references to 'val_periodic' in all driver constructors? They were used before to check for periodic meshes so that the old CPhysicalGeometry() constructor could be instantiated. This is no longer necessary now that the periodic BCs have been fixed. That variable can simply be removed from the argument list and deleted.